File: тесты/bench09_array_reduction.py
import time

def run_reduction():
    N = 100000
    arr = [0.0] * N
    i = 0
    while i < N:
        arr[i] = float((i * 3 + 7) % 1000)
        i += 1

    total = 0.0
    p = 0
    while p < 50:
        i = 0
        while i < N:
            total += arr[i]
            i += 1
        p += 1
    return total

if __name__ == "__main__":
    t0 = time.perf_counter()
    ans = run_reduction()
    t1 = time.perf_counter()
    ms = (t1 - t0) * 1000.0
    print(f"RESULT:{int(ans)}")
    print(f"TIME_MS:{ms:.2f}")

File: тесты/bench09_array_reduction.cpp
#include <iostream>
#include <vector>
#include <chrono>

int main() {
    int N = 100000;
    std::vector<double> arr(N);
    for (int i = 0; i < N; ++i) {
        arr[i] = (i * 3 + 7) % 1000;
    }

    auto t0 = std::chrono::high_resolution_clock::now();
    double total = 0.0;
    for (int pass = 0; pass < 50; ++pass) {
        for (int i = 0; i < N; ++i) {
            total += arr[i];
        }
    }
    auto t1 = std::chrono::high_resolution_clock::now();
    double ms = std::chrono::duration<double, std::milli>(t1 - t0).count();
    std::cout << "RESULT:" << (long long)total << std::endl;
    std::cout << "TIME_MS:" << ms << std::endl;
    return 0;
}
